service/node.go

```diff
@@ -420,6 +428,47 @@ func (s *service) nodeProbe(ctx context.Context) error {
 	return nil
 }
 
+func (s *service) approveSDC(opts Opts) error {
+
+	for _, systemID := range connectedSystemID {
+		system := s.systems[systemID]
+
+		if system == nil {
+			return nil
+		}
+
+		//fetch SDC details
+		sdc, err := s.systems[systemID].FindSdc("SdcGUID", opts.SdcGUID)
+		if err != nil {
+			return status.Errorf(codes.FailedPrecondition, "%s", err)
+		}
+
+		//fetch the restrictedSdcMode
+		if system.System.RestrictedSdcMode == "Guid" {
+			if !sdc.Sdc.SdcApproved {
+				resp, err := system.ApproveSdcByGUID(sdc.Sdc.SdcGUID)
+				if err != nil {
+					return status.Errorf(codes.FailedPrecondition, "%s", err)
+				}
+				Log.Infof("SDC Approved, SDC Id: %s and SDC GUID: %s", resp.SdcID, sdc.Sdc.SdcGUID)
+			} else {
+				Log.Infof("SDC already approved, SDC GUID: %s", sdc.Sdc.SdcGUID)
+			}
+		} else {
+			if !sdc.Sdc.SdcApproved {
+				return status.Errorf(codes.FailedPrecondition,
+					"Array RestrictedSdcMode is %s, driver only supports GUID RestrictedSdcMode cannot approve SDC %s",
+					system.System.RestrictedSdcMode, sdc.Sdc.SdcGUID)
```

Comment:
is it possible to fix the spacing on this print line?

Reply:
done